### ISSUE TITLE
Codable support

### ISSUE THREAD

**Issue by JoeFerrucci** [PostHog/posthog-ios#52]

**Comment by marandaneto** [comment]
@JoeFerrucci do you mean https://developer.apple.com/documentation/swift/codable ?

**Comment by JoeFerrucci** [comment]
Yes, that is the one. When I try to pass a dictionary into `capture(event:properties:)` it fails with that error.

Here is a trivial example to help highlight it

```swift
struct MyType: Codable {}

posthog?.capture("My Event", properties: ["somekey": MyType()])
```

This yields the error. Technically PostHog SDK only needs to support `Encodable`; it doesn't **need** to be `Decodable`

**Comment by marandaneto** [comment]
superseeded by https://github.com/PostHog/posthog-ios/issues/557
